storage/zynq7000-flash/flashdrv.c

```diff
@@ -11,30 +11,42 @@
  * %LICENSE%
  */
 
-#include "qspi.h"
+#include "flashdrv.h"
 #include "flashcfg.h"
+#include "qspi.h"
 
 #include <stdlib.h>
 #include <errno.h>
 #include <string.h>
 #include <sys/time.h>
+#include <sys/threads.h>
 
-#define TIMEOUT_CMD_MS 0x05
+/* Value determined empirically, contains time for communication via qspi and system calls invocations. */
+#define TIMEOUT_CMD_MS 0x50
 #define MAX_SIZE_CMD   0x20
 
+#define MTD_DEFAULT_ERASESZ 0x10000
 
-struct {
-	/* Buffers for command transactions */
-	uint8_t cmdRx[MAX_SIZE_CMD];
-	uint8_t cmdTx[MAX_SIZE_CMD];
 
+typedef struct {
 	/* Data caching in sector's buffer */
 	uint8_t *buff;
 	uint32_t buffPos;
-	uint32_t regID;
 	uint32_t sectID;
 
-	flash_info_t info;
+	/* Buffers for command transactions */
+	uint8_t cmdRx[MAX_SIZE_CMD];
+	uint8_t cmdTx[MAX_SIZE_CMD];
```

Comment:
You always memset this in the functions - it's not huge (32 bytes) so imho it would be easier and cleaner to do
`uint8_t cmdTx[SIZE_WE_REALLY_NEED] = { 0 };`
inside the function (local variable) instead of doing `hal_memset`

Just a thought, I might be wrong

Reply:
The command size depends on the flash memory. So the size is not always known a priori, some calculations are needed. It would be difficult to be compliant with our convention of declaring data at the beginning of the function.